spec.html

```diff
@@ -14621,6 +14623,64 @@ <h1>[[OwnPropertyKeys]] ( ): a normal completion containing a List of property k
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-integer-indexed-object-with-buffer-witness-records">
+        <h1>Integer-Indexed Object With Buffer Witness Records</h1>
+        <p>A <dfn variants="Integer-Indexed Object With Buffer Witness Records">Integer-Indexed Object With Buffer Witness Record</dfn> is a Record value used to encapsulate an Integer-Indexed object along with a cached byte length of the viewed buffer. It is used to help ensure there is a single shared memory read event of the byte length data block when the viewed buffer is a growable SharedArrayBuffers.</p>
+        <p>Integer-Indexed Object With Buffer Witness Records have the fields listed in <emu-xref href="#table-integer-indexed-object-with-buffer-witness-record-fields"></emu-xref>.</p>
+        <emu-table id="table-integer-indexed-object-with-buffer-witness-record-fields" caption="Integer-Indexed Object With Buffer Witness Record Fields">
+          <table>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Object]]
+              </td>
+              <td>
+                an Integer-Indexed exotic object
+              </td>
+              <td>
+                The Integer-Indexed exotic object whose buffer's byte length is loaded.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[CachedBufferByteLength]]
+              </td>
+              <td>
+                a non-negative integer
+              </td>
+              <td>
+                The byte length of the object's [[ViewedArrayBuffer]] when the Record was created.
+              </td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-makeintegerindexedobjectwithbufferwitnessrecord" type="abstract operation">
+        <h1>
+          MakeIntegerIndexedObjectWithBufferWitnessRecord (
+            _obj_: an Integer-Indexed exotic object,
+            _order_: ~SeqCst~ or ~Unordered~,
+          ): an Integer-Indexed Object With Buffer Witness Record
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Let _byteLength_ be ArrayBufferByteLength(_obj_.[[ViewedArrayBuffer]], _order_).
```

Comment:
`ArrayBufferByteLength` asserts that the buffer isn't detached, so this first needs to call IsDetachedBuffer to handle that case.

Reply:
Good catch. I've updated the buffer witness record machinery for both IIEOs and DataViews to handle detached.